kalite/topic_tools/content_recommendation.py

```diff
@@ -0,0 +1,506 @@
+'''
+All logic for generating and retriving content recommendations.
+'''
+
+from kalite.topic_tools import * 
+from kalite.main.models import ExerciseLog 
+
+
+###
+# The KING function for content recommendation. Call this single function for all 
+# of your content recommendation needs (i.e. in the API call).
+#
+# @param user: the unique user value of the current user (id/name I believe?)
+# @param current_subtopic: optional latest subtopic id that the user has worked on/completed. This
+#                          will act as a starting point for some of the algorithms (nearest neighbor, etc.)
+#
+# @return: Still deciding on this, but I assume that it will be a dictionary in the 
+#          form: { 'next_steps':[ ... ], 'explore':[ ... ], etc. } 
+###
+def get_recommendations(user=None, current_subtopic=None):
```

Comment:
I wonder how much useful work this wrapper function is doing. It might be better to just define your lower level functions that you have below, and let this level be handled by @sialan's API.

The reason I suggest this, is that if we are going to allow filtering in the API to *only* retrieve 'resume' or 'next_steps' then it would be helpful to call those functions in a more granular way in the API.